x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/categorization/GrokPatternCreator.java

```diff
@@ -124,7 +124,11 @@ public static String findBestGrokMatchFromExamples(String jobId, String regex, C
                 // the message was very long, and the example was truncated.  In this
                 // case we will have appended an ellipsis to indicate truncation.
                 assert example.endsWith("...") : exampleProcessor.pattern() + " did not match non-truncated example " + example;
-                logger.warn("[{}] Pattern [{}] did not match example [{}]", jobId, exampleProcessor.pattern(), example);
+                if (example.endsWith("...")) {
+                    logger.trace("[{}] Pattern [{}] did not match truncated example", jobId, exampleProcessor.pattern());
```

Comment:
```suggestion
                    logger.trace(() -> new ParameterizedMessage("[{}] Pattern [{}] did not match truncated example", jobId, exampleProcessor.pattern()));
```

Otherwise, even with trace logging turned off, it could be that we create a string with the pattern. I am not sure how often that occurs, but it is good practice to use a supplier in `trace` IMO.